lib/core/errors.h

```diff
@@ -11,10 +11,14 @@
 #include <signal.h>
 
 #ifdef __cplusplus
+#ifdef __clang__
+#include <string.h>
```

Comment:
Why is this needed?   Is there a way we can setup these includes such that it works with all compilers?

Reply:
If I include `cstring` compiling with `clang` I get errors:
```
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/cstring:100:3: error: conflicting types for 'memchr'
  memchr(void* __s, int __c, size_t __n)
  ^
/usr/include/string.h:90:14:
:/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/cstring: 100:3: error: conflicting types for 'memchr'
  memchr(void* __s, int __c, size_t __n)
  ^
/usr/include/string.h:90:14: note: previous declaration is here
extern void *memchr (const void *__s, int __c, size_t __n)
```

Reply:
I think the problem is that `<cstring>` and `<string.h>` are the C++ and C versions of the library respectively.    If you look above on line 9, `<string.h>` always gets included, so also including `<cstring>` breaks things.  I'm guessing compilers like `gcc` just pick one for you.  The better solution would seem to be to only include `<string.h>` in the `else` clause of the `#ifdef __cplusplus`, that way one of the two is included, and we don't have to try and also select for compilers.